pallets/rc-migrator/src/lib.rs

```diff
@@ -364,10 +369,64 @@ pub mod pallet {
 
 	impl<T: Config> Pallet<T> {
 		/// Execute a stage transition and log it.
-		fn transition(new_stage: MigrationStage<T::AccountId>) {
+		fn transition(new: MigrationStage<T::AccountId>) {
 			let old = RcMigrationStage::<T>::get();
-			RcMigrationStage::<T>::put(&new_stage);
-			log::info!(target: LOG_TARGET, "[Block {:?}] Stage transition: {:?} -> {:?}", frame_system::Pallet::<T>::block_number(), old, new_stage);
+			RcMigrationStage::<T>::put(&new);
+			log::info!(target: LOG_TARGET, "[Block {:?}] Stage transition: {:?} -> {:?}", frame_system::Pallet::<T>::block_number(), &old, &new);
+			Self::deposit_event(Event::StageTransition { old, new });
+		}
+
+		/// Split up the items into chunks of `MAX_MSG_SIZE` and send them as separate XCM
+		/// transacts.
+		///
+		/// This is done to avoid exceeding the XCM message size limit.
+		pub fn send_chunked_xcm<E: Encode>(
+			mut items: Vec<E>,
+			create_call: impl Fn(Vec<E>) -> types::AhMigratorCall<T>,
+		) -> Result<(), Error<T>> {
+			const MAX_MSG_SIZE: u32 = 50_000; // Soft message size limit. Hard limit is about 64KiB
+
+			while !items.is_empty() {
+				let mut remaining_size: u32 = MAX_MSG_SIZE;
+				let mut batch = Vec::new();
+
+				while !items.is_empty() {
+					// Order does not matter, so we take from the back as optimization
```

Comment:
maybe worth mentioning in the doc

Reply:
Oh yea, I will reverse the vector first to get rid of this.